batch.go

```diff
@@ -0,0 +1,273 @@
+	CancelledAt      *int               `json:"cancelled_at"`
+	RequestCounts    BatchRequestCounts `json:"request_counts"`
+	Metadata         map[string]any     `json:"metadata"`
+}
+
+type BatchRequestCounts struct {
+	Total     int `json:"total"`
+	Completed int `json:"completed"`
+	Failed    int `json:"failed"`
+}
+
+type CreateBatchRequest struct {
+	InputFileID      string         `json:"input_file_id"`
+	Endpoint         BatchEndpoint  `json:"endpoint"`
+	CompletionWindow string         `json:"completion_window"`
+	Metadata         map[string]any `json:"metadata"`
+}
+
+type BatchResponse struct {
+	httpHeader
+	Batch
+}
+
+var ErrUploadBatchFileFailed = errors.New("upload batch file failed")
+
+// CreateBatch — API call to Create batch.
+func (c *Client) CreateBatch(
+	ctx context.Context,
+	request CreateBatchRequest,
+) (response BatchResponse, err error) {
+	if request.CompletionWindow == "" {
+		request.CompletionWindow = "24h"
+	}
+
+	req, err := c.newRequest(ctx, http.MethodPost, c.fullURL(batchesSuffix), withBody(request))
+	if err != nil {
+		return
+	}
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+type UploadBatchFileRequest struct {
+	FileName string
+	Lines    []BatchLineItem
+}
+
+func (r *UploadBatchFileRequest) MarshalJSONL() []byte {
+	buff := bytes.Buffer{}
+	for i, line := range r.Lines {
+		if i != 0 {
+			buff.Write([]byte("\n"))
+		}
+		buff.Write(line.MarshalBatchLineItem())
+	}
+	return buff.Bytes()
+}
+
+func (r *UploadBatchFileRequest) AddChatCompletion(customerID string, body ChatCompletionRequest) {
+	r.Lines = append(r.Lines, BatchChatCompletionRequest{
+		CustomID: customerID,
+		Body:     body,
+		Method:   "POST",
+		URL:      BatchEndpointChatCompletions,
+	})
+}
+
+func (r *UploadBatchFileRequest) AddCompletion(customerID string, body CompletionRequest) {
+	r.Lines = append(r.Lines, BatchCompletionRequest{
+		CustomID: customerID,
+		Body:     body,
+		Method:   "POST",
+		URL:      BatchEndpointCompletions,
+	})
+}
+
+func (r *UploadBatchFileRequest) AddEmbedding(customerID string, body EmbeddingRequest) {
+	r.Lines = append(r.Lines, BatchEmbeddingRequest{
+		CustomID: customerID,
+		Body:     body,
+		Method:   "POST",
+		URL:      BatchEndpointEmbeddings,
+	})
+}
+
+// UploadBatchFile — upload batch file.
+func (c *Client) UploadBatchFile(ctx context.Context, request UploadBatchFileRequest) (File, error) {
+	if request.FileName == "" {
+		request.FileName = "@batchinput.jsonl"
+	}
+	return c.CreateFileBytes(ctx, FileBytesRequest{
+		Name:    request.FileName,
+		Bytes:   request.MarshalJSONL(),
+		Purpose: PurposeBatch,
+	})
+}
+
+type CreateBatchWithUploadFileRequest struct {
+	Endpoint         BatchEndpoint  `json:"endpoint"`
+	CompletionWindow string         `json:"completion_window"`
+	Metadata         map[string]any `json:"metadata"`
+	UploadBatchFileRequest
+}
+
+// CreateBatchWithUploadFile — API call to Create batch with upload file.
+func (c *Client) CreateBatchWithUploadFile(
```

Comment:
I'm not sure we need this method. Do you think it would be used in >90% of cases when people use batches?

An alternative would be to have a README example of how to upload file and create a batch

Reply:
I think providing this method would reduce complexity, and I have provided an example: https://github.com/eiixy/monorepo/blob/master/example/openai/main.go